src/Microsoft.Data.SqlClient/add-ons/AzureKeyVaultProvider/LocalCache.cs

```diff
@@ -77,9 +77,12 @@ internal TValue GetOrCreate(TKey key, Func<TValue> createItem)
                     AbsoluteExpirationRelativeToNow = TimeToLive
                 };
 
+                AKVEventSource.Log.TryTraceEvent("Encryption key added to local cache.");
+
                 _cache.Set(key, cacheEntry, cacheEntryOptions);
             }
 
+            AKVEventSource.Log.TryTraceEvent("Encryption key entry fetched from local cache.");
```

Comment:
This will be logged even if the entry did not exist in the cache.

Reply:
Yep - it will be added above to cache above so its safe to say. I added that knowingly.

Reply:
If there is a cache miss and `Func<TValue> createItem` is used to create the cache entry, the event implies that the entry was fetched from the cache, i.e. that it already existed.

Reply:
I can handle that, but it will be preceded with "new cache entry added" message so that would imply a new entry was  fetched and added to cache, I'm fine eitherway.

Reply:
Sure, that makes sense.